frontend/src/components/UserSettings/UserName.spec.js

```diff
@@ -0,0 +1,157 @@
+        await wrapper.find('input').setValue('petra')
+        await wrapper.find('svg.bi-x-circle').trigger('click')
+        expect(wrapper.findAll('div.col').at(2).text()).toBe('peter')
+      })
+
+      it('has a submit button', () => {
+        expect(wrapper.find('button[type="submit"]').exists()).toBeTruthy()
+      })
+
+      it('does not enable submit button when data is not changed', async () => {
+        await wrapper.find('form').trigger('keyup')
+        expect(wrapper.find('button[type="submit"]').attributes('disabled')).toBe('disabled')
+      })
+
+      describe('successfull submit', () => {
+        beforeEach(async () => {
+          mockAPIcall.mockResolvedValue({
+            data: {
+              updateUserInfos: {
+                validValues: 3,
+              },
+            },
+          })
+          jest.clearAllMocks()
+          await wrapper.find('input').setValue('petra')
+          await wrapper.find('form').trigger('keyup')
+          await wrapper.find('button[type="submit"]').trigger('click')
+          await flushPromises()
+        })
+
+        it('calls the API', () => {
+          expect(mockAPIcall).toBeCalledWith(
+            expect.objectContaining({
+              variables: {
+                alias: 'petra',
+              },
+            }),
+          )
+        })
+
+        it('commits username to store', () => {
+          expect(storeCommitMock).toBeCalledWith('username', 'petra')
+        })
+
+        it('toasts a success message', () => {
+          expect(toastSuccessSpy).toBeCalledWith('settings.username.change-success')
+        })
+
+        it('has an edit button again', () => {
+          expect(wrapper.find('svg.bi-pencil').exists()).toBeTruthy()
+        })
+      })
+
+      describe('submit results in server error', () => {
+        beforeEach(async () => {
+          mockAPIcall.mockRejectedValue({
+            message: 'Error',
+          })
+          jest.clearAllMocks()
+          await wrapper.find('input').setValue('petra')
+          await wrapper.find('form').trigger('keyup')
+          await wrapper.find('button[type="submit"]').trigger('click')
+          await flushPromises()
+        })
+
+        it('calls the API', () => {
+          expect(mockAPIcall).toBeCalledWith(
+            expect.objectContaining({
+              variables: {
+                alias: 'petra',
+              },
+            }),
+          )
+        })
+
+        it('toasts an error message', () => {
+          expect(toastErrorSpy).toBeCalledWith('Error')
+        })
+      })
+    })
+
+    describe('no username in store', () => {
+      beforeEach(() => {
+        mocks.$store.state.username = null
+        wrapper = Wrapper()
+      })
+
+      it('displays an information why to enter a username', () => {
+        expect(wrapper.findAll('div.col').at(2).text()).toBe('settings.username.no-username')
+      })
+    })
+  })
+})
```

Comment:
You also do not test the validation rules(?), which are to what I found not validating as expected - it might be worth an issue to find a way to test them.

Reply:
First is a nice to have, but out of scope of this PR. Second, we can test the validation by setting not valid values in the input fields. I am not sure, if the messages are visible in the unit tests. I suggest a follow ticket if needed